poly-org.el

```diff
@@ -42,7 +42,18 @@
 (define-obsolete-variable-alias 'pm-inner/org 'poly-org-innermode "v0.2")
 
 (defun poly-org-mode-matcher ()
-  (when (re-search-forward "#\\+begin_src +\\([^ \t\n]+\\)" (point-at-eol) t)
+  (when (re-search-forward
+         ;; there is no exhaustive list of block names in org-mode source
+         ;; the most complete list would be
+         ;; the default value of org-structure-template-alist
+         ;; we took everything listed there into account
+         ;; it is known however there's begin_latex and who knows what else
+         ;; that should be given yet more special treatment
+         ;; Last but not least, we repeat this list thrice in this file
+         ;; which is horrendous and something has to be done about it
+         (rx "#+begin_" (or "src" "example" "export")
```

Comment:
Frankly, I don't like `rx`. Rexexp has been here forever and is the same across languages, plus it's more concise. There is little point in re-inventing the wheel.

Reply:
Alas, we won't reach agreement on this.